ecosystem/indexer-grpc/indexer-grpc-file-store/src/processor.rs

```diff
@@ -81,145 +79,194 @@ impl Processor {
             cache_operator,
             file_store_operator,
             cache_chain_id,
-            enable_verbose_logging,
+            enable_expensive_logging,
         })
     }
 
-    // Starts the processing.
+    /// Starts the processing. The steps are
+    /// 1. Check chain id at the beginning and every step after
+    /// 2. Get the batch start version from file store metadata
+    /// 3. Start loop
+    ///   3.1 Check head from cache, decide whether we need to parallel process or just wait
+    ///   3.2 If we're ready to process, create max of 10 threads and fetch / upload data
+    ///   3.3 Update file store metadata at the end of a batch
     pub async fn run(&mut self) -> Result<()> {
         let cache_chain_id = self.cache_chain_id;
 
-        // If file store and cache chain id don't match, return an error.
-        let metadata = self
-            .file_store_operator
-            .create_default_file_store_metadata_if_absent(cache_chain_id)
-            .await
-            .context("Metadata did not match.")?;
+        let mut batch_start_version =
+            if let Some(metadata) = self.file_store_operator.get_file_store_metadata().await {
+                anyhow::ensure!(metadata.chain_id == cache_chain_id, "Chain ID mismatch.");
+                metadata.version
+            } else {
+                0
```

Comment:
maybe upload metadata file here if it doesn't exist

Reply:
We fixed this in the next PR.

Reply:
in the compression PR?

Reply:
can you just include it in this PR. it's a 1 line change. the next PR doesn't upload it in the right spot.

Reply:
nvm i get it